Tests

Arkkienkeli opened the pull request:
All tests pass now, but probably will require more work in future.

jccaicedo commented on deepprofiler/dataset/image_dataset.py:
```diff
@@ -29,12 +28,14 @@ def load_loc(self, params):
         # Load cell locations for one image
         i, config = params
         loc = deepprofiler.imaging.boxes.get_locations(self.keys[i], config)
+        print('locations', loc)
```
Remove these prints!

Arkkienkeli replied:
Fixed.

jccaicedo commented on deepprofiler/dataset/image_dataset.py:
```diff
@@ -29,12 +28,14 @@ def load_loc(self, params):
         loc["Target"] = self.targets[i][0]
         loc["Outlines"] = self.outlines[i]
         print("Image", i, ":", len(loc), "cells", end="\r")
+        print('locations', loc)
```
Remove prints

Arkkienkeli replied:
Fixed.

jccaicedo commented on tests/deepprofiler/learning/test_profiling.py:
```diff
@@ -120,11 +120,12 @@ def profile(config, dataset):
-    prof = deepprofiler.learning.profiling.Profile(config, dataset)
+def test_init(config, dataset, locations):
+    metadata = deepprofiler.dataset.image_dataset.read_dataset(config)
+    prof = deepprofiler.learning.profiling.Profile(config, metadata)
```
Why was the dataset object not used here?

Arkkienkeli replied:
Fixed.

jccaicedo commented on tests/plugins/crop_generators/test_autoencoder_crop_generator.py:
```diff
@@ -62,13 +61,29 @@ def metadata(out_dir, make_struct):
 
 @pytest.fixture(scope="function")
-def dataset(metadata, out_dir, config, make_struct):
+def dataset(metadata, out_dir, config, make_struct, locations):
```
Locations doesn't seem to be used in this function.

Arkkienkeli replied:
They have to be initilized in order to make ```dset.prepare_training_locations()``` call work.

jccaicedo commented on tests/plugins/crop_generators/test_mixup_crop_generator.py:
```diff
@@ -56,13 +58,28 @@ def metadata(config, make_struct):
     meta.splitMetadata(train_rule, val_rule)
     return meta
 
+@pytest.fixture(scope="function")
```
This piece of code seems to be reused over and over. Can we generalize it somehow?

Arkkienkeli replied:
Probably we could make shared fixtures, as ```locations``` is not the only one fixture we use over and over. https://stackoverflow.com/questions/46005796/reuse-pytest-fixtures

Arkkienkeli commented:
@jccaicedo Could you look at the updates? 
Moved out fixtures. 
Also in ```test_mixup_crop_generator::test_start``` I changed assert, seems right to me, but not 100% sure. 
```test_crop_generator_build_augmentation_graph``` does not seem to test anything (```assert``` is missing at the end, this test does not pass, nevertheless it was like this) do we need this test at all?
